edx_name_affirmation/tests/test_handlers.py

```diff
@@ -32,6 +32,40 @@ def setUp(self):  # pylint: disable=super-method-not-called
         self.proctoring_attempt_id = 2222222
 
 
+@ddt.ddt
+class PostSaveVerifiedNameTests(SignalTestCase):
+    """
+    Tests for the post_save handler on the VerifiedName model.
+    """
+
+    @ddt.data(
+      (VerifiedNameStatus.PENDING, False),
+      (VerifiedNameStatus.SUBMITTED, False),
+      (VerifiedNameStatus.APPROVED, True),
+      (VerifiedNameStatus.DENIED, False)
+    )
+    @ddt.unpack
+    def test_post_save_verified_name_approved(self, status, should_send):
+        """
+        Test that VERIFIED_NAME_APPROVED should only send if the status is changed to approved.
+        """
+        with patch('edx_name_affirmation.signals.VERIFIED_NAME_APPROVED.send') as mock_signal:
+            verified_name_obj = VerifiedName.objects.create(
+                user=self.user,
+                verified_name='Jonathan Doe',
+                profile_name=self.profile_name,
+                verification_attempt_id=self.idv_attempt_id
+            )
+            verified_name_obj.status = status
+            verified_name_obj.save()
+
+            self.assertEqual(mock_signal.called, should_send)
+            if should_send:
+                mock_signal.assert_called_with(
+                    sender='name_affirmation', user_id=self.user.id, profile_name=self.profile_name
+                )
+
```

Comment:
Would it be possible to add an additional test case here to make sure that if we update the status the IDV signal handler that the `VERIFIED_NAME_APPROVED` is also sent? 

I'm not sure if there's anything we need to watch for with bulk updates to a model (which may happen in the IDV signal handler), and if those will trigger a post_save signal receiver.

Reply:
I'll add it to one of the test cases for the IDV signal. It should trigger any time a `VerifiedName` is created/updated, but only if the status on the entry after being saved is 'approved'.

Reply:
Ah, you're right, it doesn't seem to work with bulk updates - let me take a look

Reply:
For anyone looking at this: `update()` is converted directly into a SQL statement, rather than calling `save()`, so any `pre_save` or `post_save` signals aren't triggered. https://docs.djangoproject.com/en/dev/topics/db/queries/#updating-multiple-objects-at-once